src/PublisherMethods.js

```diff
@@ -103,5 +105,37 @@ module.exports = {
         _.nextTick(function() {
             me.trigger.apply(me, args);
         });
-    }
+    },
+
+    /**
+     * Returns a Promise for the triggered action
+     */
+    triggerPromise: function(){
+        var me = this;
+        var args = arguments;
+
+        var canHandlePromise =
+            this.children.indexOf('completed') >= 0 &&
+            this.children.indexOf('failed') >= 0;
+
+        if (!canHandlePromise){
+            throw new Error('Publisher must have "completed" and "failed" child publishers');
+        }
+
+        var promise = new Promise(function(resolve, reject) {
```

Comment:
Yeah it should be a factory method instead since the native `Promise` is a constructor while `bluebird` and `q.js` have factory methods to create promises. So a `Reflux.setPromiseFactory` would be a sane approach so you can do this:

```javascript
// native. maybe default?
Reflux.setPromiseFactory(function(cb){
  if (!Promise) {
    throw new Error('Promise library or shim is missing');
  }
  return new Promise(cb);
});
// bluebird
Reflux.setPromiseFactory(require('bluebird'));
// q
Reflux.setPromiseFactory(require('q').Promise);
```